parlai/tasks/insuranceqa/build.py

```diff
@@ -11,99 +11,173 @@
+
+        for line in lines:
+            fields = line.rstrip("\n").split("\t")
+            if dtype == "train":
+                assert len(fields) == 2, "data file (%s) corrupted." % inpath
+                s_q_wids, s_good_aids = fields
+
+                q = cls.wids2sent(s_q_wids.split(), d_vocab)
+                good_ans = [d_label_answer[aid_] for aid_ in s_good_aids.split()]
+                # save good answers (train only)
+                s = '1 ' + q + '\t' + "|".join(good_ans)
+                fout.write(s + '\n')
+            else:
+                assert len(fields) == 3, "data file (%s) corrupted." % inpath
+                s_good_aids, s_q_wids, s_bad_aids = fields
+
+                q = cls.wids2sent(s_q_wids.split(), d_vocab)
+                good_ans = [d_label_answer[aid_] for aid_ in s_good_aids.split()]
+                bad_ans = [d_label_answer[aid_] for aid_ in s_bad_aids.split()]
+                # save good answers and candidates
+                s = '1 ' + q + '\t' + "|".join(good_ans) + '\t\t' + "|".join(good_ans + bad_ans)
+                fout.write(s + '\n')
+        fout.close()
+
+
+class ParseInsuranceQAV2(ParseInsuranceQA):
+    version = "V2"
+    label2answer_fname = "InsuranceQA.label2answer.token.encoded.gz"
+    # TODO: support 500, 1000, 1500 (# of label candidates)
+    data_fnames = [
+        ("train", "InsuranceQA.question.anslabel.token.100.pool.solr.train.encoded.gz"),
+        ("valid", "InsuranceQA.question.anslabel.token.100.pool.solr.valid.encoded.gz"),
+        ("test", "InsuranceQA.question.anslabel.token.100.pool.solr.test.encoded.gz")
+    ]
+
+    @classmethod
+    def create_fb_format(cls, out_path, dtype, inpath, d_vocab, d_label_answer):
+        print('building fbformat:' + dtype)
+        fout = open(os.path.join(out_path, dtype + '.txt'), 'w')
+        lines = cls.readlines(inpath)
+
+        for line in lines:
+            fields = line.rstrip("\n").split("\t")
+            if len(fields) != 4:
+                raise ValueError("data file (%s) corrupted. Line (%s)" % (repr(line), inpath))
+            else:
+                _, s_q_wids, s_good_aids, s_bad_aids = fields
+                q = cls.wids2sent(s_q_wids.split(), d_vocab)
+                good_ans = [d_label_answer[aid_] for aid_ in s_good_aids.split()]
+                bad_ans = [d_label_answer[aid_] for aid_ in s_bad_aids.split()]
+                # save
+                s = '1 ' + q + '\t' + "|".join(good_ans) + '\t\t' + "|".join(good_ans + bad_ans)
+                fout.write(s + '\n')
+        fout.close()
 
 
 def build(opt):
     dpath = os.path.join(opt['datapath'], 'InsuranceQA')
-    version = None
 
-    if not build_data.built(dpath, version_string=version):
+    if not build_data.built(dpath):
         print('[building data: ' + dpath + ']')
-        if build_data.built(dpath):
-            # An older version exists, so remove these outdated files.
-            build_data.remove_dir(dpath)
```

Comment:
actually, these changes here just got overwritten--we updated the build functions across all tasks and would like to keep the version on the left (it allows interrupted downloads to automatically be resumed!)